### ISSUE TITLE
Transaction fee and change calculation failed... using v0.21.0 binary

### ISSUE THREAD

**Issue by SomethingUseful** [bitcoin/bitcoin#19229]
After upgrade of binary bitcoind from 0.19.1 to 0.20.0 could not process transactions. Did not see anything in release notes that would be breaking. Any idea what could be the issue? I cant find any documentation on this error message.

**Expected behavior**

Process transaction without error using "sendtoaddress"

**Actual behavior**

error code -4 "Transaction fee and change calculation failed"

**To reproduce**

All we did was upgrade binary bitcoind from 0.19.1 to v0.20.0. Nothing else changed.

**System information**

Using binary from https://bitcoincore.org/bin/bitcoin-core-0.20.0/bitcoin-0.20.0-x86_64-linux-gnu.tar.gz

On Linux Ubuntu 18.04.3 LTS on a SSD

No ERRORs exist in debug.log. Node syncs fine on v0.20.0... The following seems relevant but we saw same logs on v0.19.1.

**v0.19.1...**

2020-06-09T07:51:36Z [default wallet] Fee Calculation: Fee:2871 Bytes:136 Needed:2871 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99931: Estimation: (20593.8 - 21623.5) 97.20% 37504.3/(38582.5 0 mem 3.0 out) Fail: (19613.1 - 20593.8) 92.07% 24792.5/(26908.8 0 mem 19.2 out)
2020-06-09T07:54:32Z [default wallet] Fee Calculation: Fee:13110 Bytes:621 Needed:13110 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99931: Estimation: (20593.8 - 21623.5) 97.20% 37504.3/(38582.5 0 mem 3.0 out) Fail: (19613.1 - 20593.8) 92.07% 24792.5/(26908.8 0 mem 19.2 out)
2020-06-09T09:58:09Z [default wallet] Fee Calculation: Fee:7326 Bytes:347 Needed:7326 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99931: Estimation: (20593.8 - 21623.5) 97.39% 40008.9/(41076.7 0 mem 3.0 out) Fail: (19613.1 - 20593.8) 92.28% 25270.9/(27366.8 0 mem 19.0 out)
2020-06-09T13:22:31Z [default wallet] Fee Calculation: Fee:16956 Bytes:803 Needed:16956 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99931: Estimation: (20593.8 - 21623.5) 97.64% 43671.2/(44725.9 0 mem 3.0 out) Fail: (19613.1 - 20593.8) 92.48% 25689.0/(27759.1 0 mem 18.8 out)
2020-06-09T14:36:45Z [default wallet] Fee Calculation: Fee:54877 Bytes:2255 Needed:54877 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (23839.9 - 25031.9) 98.23% 3643.5/(3709.2 0 mem 0.0 out) Fail: (22704.7 - 23839.9) 81.84% 2485.0/(3036.6 0 mem 0.0 out)
2020-06-09T15:33:30Z [default wallet] Fee Calculation: Fee:5709 Bytes:258 Needed:5709 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (21623.5 - 22704.7) 88.84% 3368.3/(3791.3 0 mem 0.0 out) Fail: (20593.8 - 21623.5) 77.45% 3045.0/(3928.4 0 mem 3.2 out)
2020-06-09T16:06:13Z [default wallet] Fee Calculation: Fee:90020 Bytes:4068 Needed:90020 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (21623.5 - 22704.7) 88.87% 3127.1/(3518.6 0 mem 0.0 out) Fail: (20593.8 - 21623.5) 77.48% 2822.9/(3640.5 0 mem 2.9 out)
2020-06-09T18:16:54Z [default wallet] Fee Calculation: Fee:7683 Bytes:347 Needed:7683 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (21623.5 - 22704.7) 92.06% 3166.4/(3439.3 0 mem 0.0 out) Fail: (20593.8 - 21623.5) 82.08% 2307.7/(2809.5 0 mem 2.1 out)
2020-06-09T18:28:31Z [default wallet] Fee Calculation: Fee:11759 Bytes:531 Needed:11759 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (21623.5 - 22704.7) 92.58% 3152.3/(3404.9 0 mem 0.0 out) Fail: (20593.8 - 21623.5) 82.64% 2219.5/(2683.9 0 mem 2.0 out)
2020-06-09T18:53:50Z [default wallet] Fee Calculation: Fee:11716 Bytes:529 Needed:11716 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (21623.5 - 22704.7) 93.28% 3003.3/(3219.7 0 mem 0.0 out) Fail: (20593.8 - 21623.5) 83.47% 2016.6/(2414.3 0 mem 1.7 out)
2020-06-09T19:18:43Z [default wallet] Fee Calculation: Fee:4962 Bytes:224 Needed:4962 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (21623.5 - 22704.7) 93.59% 2925.7/(3126.0 0 mem 0.0 out) Fail: (20593.8 - 21623.5) 83.74% 1903.9/(2272.0 0 mem 1.6 out)

**v0.20.0 ...**

2020-06-09T20:18:06Z [default wallet] Fee Calculation: Fee:2829 Bytes:134 Needed:2829 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99931: Estimation: (20593.8 - 21623.5) 97.79% 45111.8/(46130.7 0 mem 2.9 out) Fail: (19613.1 - 20593.8) 93.00% 26822.9/(28822.8 0 mem 18.2 out)

**v0.19.1...**

2020-06-09T21:44:39Z [default wallet] Fee Calculation: Fee:7372 Bytes:349 Needed:7372 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (20593.8 - 21623.5) 91.12% 2121.8/(2327.6 0 mem 0.9 out) Fail: (19613.1 - 20593.8) 83.14% 777.5/(935.0 0 mem 0.1 out)
2020-06-09T21:45:57Z [default wallet] Fee Calculation: Fee:7330 Bytes:347 Needed:7330 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (20593.8 - 21623.5) 91.12% 2121.8/(2327.6 0 mem 0.9 out) Fail: (19613.1 - 20593.8) 83.14% 777.5/(935.0 0 mem 0.1 out)
2020-06-09T21:50:48Z [default wallet] Fee Calculation: Fee:36125 Bytes:1710 Needed:36125 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (20593.8 - 21623.5) 91.12% 2121.8/(2327.6 0 mem 0.9 out) Fail: (19613.1 - 20593.8) 83.14% 777.5/(935.0 0 mem 0.1 out)
2020-06-09T22:04:02Z [default wallet] Fee Calculation: Fee:4775 Bytes:226 Needed:4775 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (20593.8 - 21623.5) 92.53% 2369.9/(2560.4 0 mem 0.8 out) Fail: (19613.1 - 20593.8) 84.30% 783.4/(929.1 0 mem 0.1 out)

**Comment by alvinoktavbitcoin** [comment]
Mainnet or testnet ?

**Comment by alvinoktavbitcoin** [comment]
> `falldumpfee` is not a valid option, there is no way an invalid option can influence the wallet code because invalid options are not accepted when starting Bitcoin Core.

I have a same problem with this post. Yesterday I read a documentation in 0.20.0 Git, and it said that we need to add falldumpfee parameter in bitcoin.conf, I did it and it solve the problem. 
However my node is testnet. For that reason I asked whether his node is mainnet or testnet, because on the documentation the mainnet node doeesn't need to add the falldumpfee parameter

**Comment by hebasto** [comment]
@alvinoktavbitcoin 
> I have a same problem with this post. Yesterday I read a documentation in 0.20.0 Git, and it said that we need to add falldumpfee parameter in bitcoin.conf, I did it and it solve the problem.

What "documentation in 0.20.0 Git"? Mind posting a link?

```
$ git checkout v0.20.0
HEAD is now at a62f0ed64 doc: Manpages update pre-final
$ git grep falldumpfee
# nothing
```

**Comment by alvinoktavbitcoin** [comment]
> @alvinoktavbitcoin
> 
> > I have a same problem with this post. Yesterday I read a documentation in 0.20.0 Git, and it said that we need to add falldumpfee parameter in bitcoin.conf, I did it and it solve the problem.
> 
> What "documentation in 0.20.0 Git"? Mind posting a link?
> 
> ```
> $ git checkout v0.20.0
> HEAD is now at a62f0ed64 doc: Manpages update pre-final
> $ git grep falldumpfee
> # nothing
> ```

I read in https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-no
On "Test" Section

**Comment by hebasto** [comment]
@alvinoktavbitcoin Do you mean `-fallbackfee`?

**Comment by alvinoktavbitcoin** [comment]
> @alvinoktavbitcoin Do you mean `-fallbackfee`?

oh yes sorry, I just realize I wrongly type

**Comment by MarcoFalke** [comment]
If this is a bug in our code it is impossible to fix by a run time option (unless a hacky workaround is acceptable).

This should be properly investigated and fixed, ideally with a regression test.

**Comment by SomethingUseful** [comment]
So what is the fix here and why? Can someone explain "fallbackfee = 0" or provide documentation as to its purpose?

**Comment by dwjorgeb** [comment]
We're having the same issue with one stubborn transactions that won't go through.

Using `sendmany`:
` {'code': -6, 'message': 'Transaction fee and change calculation failed'}`

**Comment by dwjorgeb** [comment]
It fixed itself after a few transactions... odd

**Comment by SomethingUseful** [comment]
so we tried to upgrade again. Same exact issue. Node occasionally returns error code -4 "Transaction fee and change calculation failed". Its not failing 100%. some transactions go through fine.

**Comment by SomethingUseful** [comment]
This seems related ...

2020-07-01T06:48:48Z [default wallet] Fee Calculation: Fee:11378 Bytes:294 Needed:11127 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (36983.5 - 38832.7) 87.03% 1477.
5/(1697.0 0 mem 0.6 out) Fail: (35222.4 - 36983.5) 71.34% 665.2/(907.5 0 mem 25.1 out)

2020-07-01T06:52:15Z [default wallet] Fee Calculation: Fee:15385 Bytes:406 Needed:15368 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (36983.5 - 38832.7) 87.24% 1448.
3/(1659.5 0 mem 0.6 out) Fail: (35222.4 - 36983.5) 71.49% 645.0/(878.0 0 mem 24.1 out)

2020-07-01T07:33:18Z [default wallet] Fee Calculation: Fee:5161 Bytes:136 Needed:5152 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (36983.5 - 38832.7) 89.40% 1415.9/
(1583.3 0 mem 0.5 out) Fail: (35222.4 - 36983.5) 73.00% 551.2/(735.9 0 mem 19.1 out)

2020-07-01T13:15:44Z [default wallet] Fee Calculation: Fee:17378 Bytes:482 Needed:17378 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99520: Estimation: (35222.4 -
36983.5) 100.00% 8229.2/(8229.3 0 mem 0.0 out) Fail: (33545.1 - 35222.4) 94.96% 8901.2/(9373.3 0 mem 0.0 out)

2020-07-01T13:25:06Z [default wallet] Fee Calculation: Fee:9266 Bytes:257 Needed:9266 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99520: Estimation: (35222.4 - 36
983.5) 100.00% 8229.2/(8229.3 0 mem 0.0 out) Fail: (33545.1 - 35222.4) 94.96% 8901.2/(9373.3 0 mem 0.0 out)

2020-07-01T13:37:12Z [default wallet] Fee Calculation: Fee:11553 Bytes:315 Needed:11357 Tgt:6 (requested 6) Reason:"Conservative Double Target longer horizon" Decay 0.99520: Estimation: (35222.4 -
36983.5) 100.00% 8229.2/(8229.3 0 mem 0.0 out) Fail: (33545.1 - 35222.4) 94.96% 8901.2/(9373.3 0 mem 0.0 out)

**Comment by SomethingUseful** [comment]
**Now we are getting these on v0.19.1 too** ... :(

2020-07-01T18:13:26Z [default wallet] Fee Calculation: Fee:19277 Bytes:440 Needed:19277 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (42813 - 44953.7) 98.41% 5566.9/
(5657.0 0 mem 0.0 out) Fail: (40774.3 - 42813) 82.14% 5015.5/(5603.7 0 mem 502.2 out)

2020-07-01T18:16:36Z [default wallet] Fee Calculation: Fee:18270 Bytes:417 Needed:18270 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (42813 - 44953.7) 98.41% 5566.9/
(5657.0 0 mem 0.0 out) Fail: (40774.3 - 42813) 82.14% 5015.5/(5603.7 0 mem 502.2 out)

2020-07-01T18:17:42Z [default wallet] Fee Calculation: Fee:19277 Bytes:440 Needed:19277 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (42813 - 44953.7) 98.41% 5566.9/
(5657.0 0 mem 0.0 out) Fail: (40774.3 - 42813) 82.14% 5015.5/(5603.7 0 mem 502.2 out)

2020-07-01T18:25:14Z [default wallet] Fee Calculation: Fee:5964 Bytes:136 Needed:5964 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (42813 - 44953.7) 98.59% 5808.4/(5
891.8 0 mem 0.0 out) Fail: (40774.3 - 42813) 82.23% 4670.4/(5214.7 0 mem 464.8 out)

2020-07-01T18:27:50Z [default wallet] Fee Calculation: Fee:7361 Bytes:168 Needed:7361 Tgt:6 (requested 6) Reason:"Target 85% Threshold" Decay 0.96200: Estimation: (42813 - 44953.7) 98.73% 6207.7/(6
287.9 0 mem 0.0 out) Fail: (40774.3 - 42813) 83.63% 4959.9/(5483.6 0 mem 447.1 out)

**Comment by ashwinkesireddy** [comment]
Was getting a similar _"Fee estimation failed. Fallbackfee is disabled. Wait a few blocks or enable -fallbackfee."_ in regtest mode. Making an rpc 'settxfee' call before sending to an address on second node worked for me.

**Comment by beertastic** [comment]
I'm also getting this error using 'sentmany'
I have fallback fee=0.0002

I've also tried 0|

Still getting that error.... is there a fix in place for this at all? Cheers :)

**Comment by jbg** [comment]
I get this error using `walletcreatefundedpsbt` as well. I'm specifying a `feeRate` and `subtractFeeFromOutputs`, and the output is many times larger than the fee, so I don't see any reason it shouldn't work, but it doesn't.

**Comment by SomethingUseful** [comment]
anybody still getting this in 0.21.0 ? we decided to skip 0.20.x

**Comment by SomethingUseful** [comment]
We skipped 0.20.x and we just tried v0.21.0 and having same exact issue. About 50% of transactions fail while others go through.

RPC Call returns 500 with error code -4 "Transaction fee and change calculation failed"

**Comment by SomethingUseful** [comment]
A BitcoinCharts site shows that current transaction average fee (as of today) is...
Average transaction fee, USD | 0.00039 BTC ($14.25 USD) 0.000001 BTC/byte
Which equates to 0.001 BTC/kilobyte

The bitcoin.conf default is 0.0002 (probably from 2009?? and not expecting fees to be this expensive?).

We are setting fallbackfee=0.0006 in config now instead of 0.0002 and seeing if that helps...

Node restart debug logs shows it accepted it.
2021-02-04T16:30:40Z Config file arg: [main] fallbackfee="0.0006"

Maybe the fix is the RPC settxfee like posted above but the fallbackfee is the bandaid fix?
I will report back if any progress...

**Comment by achow101** [comment]
This error has nothing to do with fee estimation and no amount of fee estimation parameters will fix it. The error occurs in coin selection code is saying that the amount it is expecting to pay in transaction fees is not the amount that it calculates it actually needs to pay.

**Comment by achow101** [comment]
The only reasonable way I can see this happening is that the required fee rate changes between each iteration of the `CreateTransaction` loop. How this would work is that on one iteration, inputs are selected and the fee for a transaction containing those inputs is computed and stored as `nFeeRet`. It then reaches a condition where it does `pick_new_inputs = false` (either subtracting fee from outputs, or large excess fee after input selection), does another iteration of the loop without picking new inputs, but still recalculating the transaction fee. If the fee rate has changed between these two iterations (perhaps due to an active fee market), the second fee calculation could result in a higher transaction fee required than calculated in the first iteration thus resulting in `nFeeNeeded < nFeeRet` which hits the error that people are running into here.

As I noted in my last comment, #17331 should fix this by getting rid of that looping behavior.

**Comment by chappjc** [comment]
FWIW, we've also seen this with `subtractfeefromamount=true`

**Comment by SomethingUseful** [comment]
We have been running latest version for a week now and still seeing this error periodically. What can we do to bandaid/fix this? Do we have to wait 6 months for next release?

**Comment by jonatack** [comment]
Hi @SomethingUseful you can build from source at the head commit of #17331 (see the `doc/build-*` guides or [this tutorial](https://jonatack.github.io/articles/how-to-compile-bitcoin-core-and-run-the-tests) if needed) and test or run bitcoin core with that (with small amounts at first) and leave review feedback if it resolves the issue for you.

**Comment by MarcoFalke** [comment]
I think the pull fixing this is #21083, not  #17331

**Comment by SomethingUseful** [comment]
> Hi @SomethingUseful you can build from source at the head commit of #17331 (see the `doc/build-*` guides or [this tutorial](https://jonatack.github.io/articles/how-to-compile-bitcoin-core-and-run-the-tests) if needed) and test or run bitcoin core with that (with small amounts at first) and leave review feedback if it resolves the issue for you.

Tried this and it errored... anything obvious I am missing?

STEPS TO COMPILE w/ PR:
```
# apt update && apt dist-upgrade
# apt-get install build-essential libtool autotools-dev automake pkg-config bsdmainutils python3 libssl-dev libevent-dev libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libminiupnpc-dev libzmq3-dev libqt5gui5 libqt5core5a libqt5dbus5 qttools5-dev qttools5-dev-tools libprotobuf-dev protobuf-compiler git libsqlite3-dev ccache
# cd /usr/local/src
# mkdir bitcoin-0.21.0-patch
# cd bitcoin-0.21.0-patch/
# git clone https://github.com/bitcoin/bitcoin.git .
# ./contrib/install_db4.sh `pwd`

# nano .git/config
fetch = +refs/pull/*/head:refs/remotes/origin/pr/*

# git fetch origin
# git checkout pr/17331
# export BDB_PREFIX='/usr/local/src/bitcoin-0.21.0-patch/db4'
# ./autogen.sh
# ./configure BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" BDB_CFLAGS="-I${BDB_PREFIX}/include"
# make -j "$(($(nproc)+1))"
```
 
  CXX      wallet/libbitcoin_wallet_a-rpcwallet.o
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
Makefile:8894: recipe for target 'wallet/libbitcoin_wallet_a-rpcwallet.o' failed
make[2]: *** [wallet/libbitcoin_wallet_a-rpcwallet.o] Error 4
make[2]: Leaving directory '/usr/local/src/bitcoin-0.21.0-patch/src'
Makefile:14461: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/usr/local/src/bitcoin-0.21.0-patch/src'
Makefile:812: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

**Comment by MarcoFalke** [comment]
>  g++: internal compiler error: Killed (program cc1plus)

That means gcc hit OOM. Can be fixed by typing `make` (without j) or by adding memory.

**Comment by SomethingUseful** [comment]
> > g++: internal compiler error: Killed (program cc1plus)
> 
> That means gcc hit OOM. Can be fixed by typing `make` (without j) or by adding memory.

more memory worked.

**Comment by SomethingUseful** [comment]
The "STEPS TO COMPILE w/ PR" post i made above worked for us. No transactions issues since 4 days ago.

**Comment by achow101** [comment]
> I think the pull fixing this is #21083, not #17331 (Maybe either of them fixes this?)

Either should fix this issue however #21083 is a direct fix. It also avoids having similar issues in the future in a few other places.

**Comment by SomethingUseful** [comment]
i updated my comment above to use 21083 instead of 17331 in case someone else follows it.

**Comment by MarcoFalke** [comment]
DrahtBot says they are conflicting. Does the order matter in which they are merged. (Asking for the backports, if any)

**Comment by achow101** [comment]
> DrahtBot says they are conflicting. Does the order matter in which they are merged. (Asking for the backports, if any)

Only #21083 should be backported, and it should be merged first for the backport to go smoothly.

**Comment by dwjorgeb** [comment]
So, I commented on this 9 months ago, yet this issue keeps appearing, even on v21... any timetable for the fix to be included?

**Comment by vanzay** [comment]
Still appears in 0.21.1
Sending 0.042709 btc to bc1qfqeq9x5qjcra30ceqnd9h0lzvhd25hxk2p84s7 constantly fails with "Transaction fee and change calculation failed".
Logs:
2021-05-04T05:03:50Z FeeEst: 504 > 60% decay 0.99931: feerate: 4057.17 from (3920.13 - 4116.14) 83.94% 2587.5/(2668.5 0 mem 413.9 out) Fail: (3071.52 - 3920.13) 23.12% 161.2/(181.2 0 mem 515.9 out)
2021-05-04T05:03:50Z FeeEst: 48 > 60% decay 0.99520: feerate: 5059.02 from (5003.19 - 5253.35) 71.99% 2559.9/(3486.5 0 mem 69.1 out) Fail: (4764.94 - 5003.19) 53.82% 1264.4/(1905.1 0 mem 444.3 out)
2021-05-04T05:03:50Z FeeEst: 12 > 60% decay 0.96200: feerate: 4985.14 from (4764.94 - 5003.19) 82.91% 659.7/(794.3 0 mem 1.3 out) Fail: (4538.04 - 4764.94) 36.94% 29.1/(78.1 0 mem 0.7 out)
2021-05-04T05:03:50Z FeeEst: 1008 > 85% decay 0.99931: feerate: 4057.17 from (3920.13 - 4116.14) 87.85% 2642.3/(2668.5 0 mem 339.4 out) Fail: (3071.52 - 3920.13) 29.80% 173.1/(181.2 0 mem 399.6 out)
2021-05-04T05:03:50Z FeeEst: 48 > 85% decay 0.99520: feerate: 6015.9 from (5791.82 - 6081.41) 87.35% 4912.0/(5619.7 0 mem 3.8 out) Fail: (5516.02 - 5791.82) 73.78% 259.3/(349.9 0 mem 1.5 out)
2021-05-04T05:03:50Z FeeEst: 12 > 85% decay 0.96200: feerate: 5055.88 from (5003.19 - 5253.35) 97.21% 1367.4/(1406.2 0 mem 0.5 out) Fail: (4764.94 - 5003.19) 82.91% 659.7/(794.3 0 mem 1.3 out)
2021-05-04T05:03:50Z FeeEst: 3 > 60% decay 0.96200: feerate: 5055.88 from (5003.19 - 5253.35) 79.17% 1114.1/(1406.2 0 mem 0.9 out) Fail: (4764.94 - 5003.19) 52.75% 419.7/(794.3 0 mem 1.4 out)
2021-05-04T05:03:50Z FeeEst: 6 > 85% decay 0.96200: feerate: 5055.88 from (5003.19 - 5253.35) 96.81% 1361.8/(1406.2 0 mem 0.6 out) Fail: (4764.94 - 5003.19) 80.45% 640.1/(794.3 0 mem 1.3 out)
2021-05-04T05:03:50Z FeeEst: 12 > 95% decay 0.96200: feerate: 48449.2 from (47201.4 - 49561.4) 100.00% 597.8/(597.8 0 mem 0.0 out) Fail: (44953.7 - 47201.4) 93.69% 265.6/(283.5 0 mem 0.0 out)
2021-05-04T05:03:50Z FeeEst: 504 > 60% decay 0.99931: feerate: 4057.17 from (3920.13 - 4116.14) 83.94% 2587.5/(2668.5 0 mem 413.9 out) Fail: (3071.52 - 3920.13) 23.12% 161.2/(181.2 0 mem 515.9 out)
2021-05-04T05:03:50Z FeeEst: 48 > 60% decay 0.99520: feerate: 5059.02 from (5003.19 - 5253.35) 71.99% 2559.9/(3486.5 0 mem 69.1 out) Fail: (4764.94 - 5003.19) 53.82% 1264.4/(1905.1 0 mem 444.3 out)
2021-05-04T05:03:50Z FeeEst: 12 > 60% decay 0.96200: feerate: 4985.14 from (4764.94 - 5003.19) 82.91% 659.7/(794.3 0 mem 1.3 out) Fail: (4538.04 - 4764.94) 36.94% 29.1/(78.1 0 mem 0.7 out)
2021-05-04T05:03:50Z FeeEst: 1008 > 85% decay 0.99931: feerate: 4057.17 from (3920.13 - 4116.14) 87.85% 2642.3/(2668.5 0 mem 339.4 out) Fail: (3071.52 - 3920.13) 29.80% 173.1/(181.2 0 mem 399.6 out)
2021-05-04T05:03:50Z FeeEst: 48 > 85% decay 0.99520: feerate: 6015.9 from (5791.82 - 6081.41) 87.35% 4912.0/(5619.7 0 mem 3.8 out) Fail: (5516.02 - 5791.82) 73.78% 259.3/(349.9 0 mem 1.5 out)
2021-05-04T05:03:50Z FeeEst: 12 > 85% decay 0.96200: feerate: 5055.88 from (5003.19 - 5253.35) 97.21% 1367.4/(1406.2 0 mem 0.5 out) Fail: (4764.94 - 5003.19) 82.91% 659.7/(794.3 0 mem 1.3 out)

**Comment by chappjc** [comment]
Sorry to report that we've hit the error multiple times with 0.21.1 as well.

**Comment by jholdstock** [comment]
I'm also seeing this still on 0.21.1

**Comment by dwjorgeb** [comment]
Still happening on v21.1 for us too. Please re-open this issue

**Comment by meshcollider** [comment]
This should have been fixed in v22 and master by #17331. Closing again, please comment if you hit this in v22 onward.

**Comment by chappjc** [comment]
Recently seen on regtest with v23.
